File: Lendship.Frontend/src/app/models/evaluation.ts
import {User} from "./user";

export interface IEvaluationAdvertiser {
  userFrom: User | undefined,
  userTo: User,
  advertisementId: number,
  flexibility: number,
  reliability: number,
  qualityOfProduct: number,
  comment: string,
  isAnonymous: boolean
}

export class EvaluationAdvertiser implements IEvaluationAdvertiser {
  userFrom: User | undefined = undefined
  userTo: User
  advertisementId: number
  reservationId: number
  flexibility: number
  reliability: number
  qualityOfProduct: number
  comment: string
  isAnonymous: boolean

  constructor(userTo: User, advertisementId: number, reservationId: number, flexibility: number, reliability: number, qualityOfProduct: number, comment: string, isAnonymous: boolean) {
    this.userTo = userTo;
    this.advertisementId = advertisementId;
    this.reservationId = reservationId;
    this.flexibility = flexibility;
    this.reliability = reliability;
    this.qualityOfProduct = qualityOfProduct;
    this.comment = comment;
    this.isAnonymous =isAnonymous;
  }
}

export interface IEvaluationLender {
  userFrom: User | undefined,
  userTo: User,
  advertisementId: number,
  flexibility: number,
  reliability: number,
  qualityAtReturn: number,
  comment: string,
  isAnonymous: boolean
}

export class EvaluationLender implements IEvaluationLender {
  userFrom: User | undefined = undefined
  userTo: User
  advertisementId: number
  reservationId: number
  flexibility: number
  reliability: number
  qualityAtReturn: number
  comment: string
  isAnonymous: boolean

  constructor(userTo: User, advertisementId: number, reservationId: number, flexibility: number, reliability: number, qualityAtReturn: number, comment: string, isAnonymous: boolean) {
    this.userTo = userTo;
    this.advertisementId = advertisementId;
    this.reservationId = reservationId
    this.flexibility = flexibility;
    this.reliability = reliability;
    this.qualityAtReturn = qualityAtReturn;
    this.comment = comment;
    this.isAnonymous =isAnonymous;
  }
}

File: Lendship.Backend/Models/Evaluation.cs
using Lendship.Backend.Authentication;
using System;
using System.ComponentModel.DataAnnotations;

namespace Lendship.Backend.Models
{
    public class Evaluation
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public ApplicationUser UserFrom { get; set; }

        [Required]
        public  ApplicationUser UserTo { get; set; }

        [Required]
        public int AdvertisementId { get; set; }

        [Required]
        public int ReservationId { get; set; }

        [Required]
        public int Flexibility { get; set; }

        [Required]
        public int Reliability { get; set; }

        public string Comment { get; set; }

        [Required]
        public bool IsAnonymous { get; set; }

        [Required]
        public DateTime Creation { get; set; }
    }
}
